samples/go/csv/csv-invert/main.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/attic-labs/noms/go/config"
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/noms/go/datas"
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/go/util/profile"
+	flag "github.com/juju/gnuflag"
+)
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [options] <dataset-to-invert> <output-dataset>\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+
+	profile.RegisterProfileFlags(flag.CommandLine)
+	flag.Parse(true)
+
+	if flag.NArg() != 2 {
+		flag.Usage()
+		return
+	}
+
+	cfg := config.NewResolver()
+	inDB, inDS, err := cfg.GetDataset(flag.Arg(0))
+	d.CheckError(err)
+	defer inDB.Close()
+
+	head, present := inDS.MaybeHead()
+	if !present {
+		d.CheckErrorNoUsage(fmt.Errorf("The dataset %s has no head", flag.Arg(0)))
+	}
+	v := head.Get(datas.ValueField)
+	l, isList := v.(types.List)
+	if !isList {
```

Comment:
We'll need to also handle maps to support index by primary key. This isn't necessary for the taxi benchmark, so it can wait.